src/frequenz/sdk/microgrid/_power_distributing/_component_status/_battery_status_tracker.py

```diff
@@ -262,7 +262,9 @@ async def _run(
         inverter = inverter_receiver
         set_power_result = set_power_result_receiver
 
-        while True:
+        # To prevent entering a shutdown-loop, we explicitly check if there is
+        # an eventloop running
+        while asyncio.get_event_loop().is_running():
```

Comment:
Shouldn't this stop when there is an exception (like there is no loop)? Or is this being run in some other `run_forever()` or something like that?

Reply:
But I think it is fine, if there is no loop an exception is raised in `select()`, the issue is we are blindly looping on broken code forever.